source/common/http/http1/codec_impl.cc

```diff
@@ -693,26 +683,38 @@ void ServerConnectionImpl::handlePath(HeaderMap& headers, unsigned int method) {
   headers.setHost(absolute_url.host_and_port());
 
   headers.setPath(absolute_url.path_and_query_params());
-  active_request_->request_url_.clear();
+  active_request_.value().request_url_.clear();
 }
 
 int ServerConnectionImpl::onHeadersComplete() {
   // Handle the case where response happens prior to request complete. It's up to upper layer code
   // to disconnect the connection but we shouldn't fire any more events since it doesn't make
   // sense.
-  if (active_request_) {
+  if (active_request_.has_value()) {
     auto& headers = absl::get<RequestHeaderMapPtr>(headers_or_trailers_);
     ENVOY_CONN_LOG(trace, "Server: onHeadersComplete size={}", connection_, headers->size());
     const char* method_string = http_method_str(static_cast<http_method>(parser_.method));
 
+    if (!handling_upgrade_ && connection_header_sanitization_ && headers->Connection()) {
```

Comment:
This is a slight functional change. In the old code, sanitizing never happened if there was an upgrade attempt in the request. Now it happens if the upgrade was cleaned of h2c. I did this for simplicity because I can't see any reason for the previous behavior. Please let me know if there is something I am missing.

Reply:
cc @yanavlasov also who I think was heavily involved in the original review on the connection sanitizing PR.

Reply:
My guess is it was more because the upgrade case is finicky about upgrade headers, but keep-alive and upgrade are special cased so no worries there.

This is the sort of change we'd runtime guard in-house, and while I suspect it's too much of a PITA to guard the whole refactor, maybe at least call out the functional change in release notes?

Reply:
Sure will add a release note. IMO this change is actually safer to some extent security wise so I would call it an improvement.